drf_auto_endpoint/metadata.py

```diff
@@ -51,24 +51,26 @@ def determine_metadata(self, request, view):
         if view.__class__.__name__ in root_view_names or view in root_view_names:
             return self.root_metadata(metadata, view)
 
+        serializer = view.get_serializer_class()
+
         try:
             serializer_instance = view.get_serializer()
         except Exception:
             # Custom viewset is expecting something we can't guess
-            serializer_instance = view.get_serializer_class()()
+            serializer_instance = serializer()
         endpoint = None
         if hasattr(view, 'endpoint'):
             endpoint = view.endpoint
         else:
-            if hasattr(view.get_serializer_class().Meta, 'model'):
+            if hasattr(serializer.Meta, 'model'):
                 from .endpoints import Endpoint
-                endpoint = Endpoint(view.get_serializer_class().Meta.model, viewset=view)
+                endpoint = Endpoint(serializer.Meta.model, viewset=view)
 
         adapter = import_string(settings.METADATA_ADAPTER)()
         if endpoint is None:
             fields_metadata = []
 
-            for field in view.get_serializer_class().Meta.fields:
+            for field in serializer().fields.keys():
```

Comment:
Since this line is being modified anyways, `serializer_instance` is already initialized, lets use that instead